### ISSUE TITLE
Inpainting task

### ISSUE THREAD

**Issue by balooii** [Dixin-Lab/BFVR-STC#6]
I'm wondering if you've tried to train a model for Inpainting? It wasn't mentioned in the paper but its mentioned in the code so maybe you've got some experience with it.

Training VQGAN/stage1 worked just fine but then when training stage2 on an Inpainting dataset I struggle to get anything useful. I've tried both  the bfvr and de-flickering options and trained each for 30k steps but the loss curves aren't changing much after 10k steps.

Will experiment a bit more with it but if you have something to share I'd appreciate it.

**Comment by hhhh1138** [comment]
I've tried training the inpainting task on a video face dataset before and couldn't get good inpainting results either. The way I set up the mask is to pick a random position in the video ((x,y)|0<x<W-d,0<y<H-d) and add a square mask of random size d (d in [0.1W, 0.4W]) to all frames. The trained inpainting model works poorly. I think the structural design of this model may not be able to support full video inpainting. You can try to train and inference by adding masks to only partial frames of the original video, this task should be easier.

**Comment by balooii** [comment]
I've tried a similar setup: Random rectangular mask with sides (width/height) between 0.1 and 0.2 centered in the middle of the frame for all frames.

Would be a bummer if it doesn't work. What I like most about your project is that its inference memory and compute requirements make it potentially usable on high-end consumer hardware like the RTX3090 i've been trying it on. All other diffusion-based approaches I've seen so far are orders of magnitude too slow or need datacenter GPUs to run so I think your efficient approach has a lot of potential.

**Comment by hhhh1138** [comment]
You can also try some other BFVR methods, such as PGTFormer and KEEP, they are also VQGAN-based methods and can run under 24GB VRAM hardware.🔎

**Comment by balooii** [comment]
Thank you, I will look further into those 👍 . Unfortunately booth don't seem to have published their training code so would be tricky to fine-tune it on my dataset or experiment.
